src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,87 @@
-export const Pagination = () => {};
+import React from 'react';
+
+type PaginationProps = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const pageNumbers: number[] = [];
+
+  for (let i = 1; i <= totalPages; i += 1) {
```

Comment:
Avoid using a `for` loop to create a new array with each element increased by one. Instead, use `map`.